File: Python Snippets/binary_search.py
def binary_search(nums, target):
  '''
  given an array of numbers and a target
  number, return the targets index
  '''
  low = 0
  high = len(nums) - 1

  while low <= high:
    mid = (low + high) // 2

    if nums[mid] == target:
      return mid
    elif nums[mid] < target:
      low = mid + 1
    else:
      high = mid - 1

  return -1

sample_numbers = [10, 5, 3, -4, 15, 2, 1, 15]
index = binary_search(sample_numbers, 0)

File: R Snippets/binary_search.r
binary_search <- function(nums, target) {
    low <- 0
    high <- length(nums)

    while (low <= high) {
        mid <- (low + high) %/% 2

        if (nums[mid] == target) {
            return(mid)
        } else if (nums[mid] < target) {
            low <- mid + 1
        } else {
            high <- mid - 1
        }
    }

    return(-1)
}

sample_vector <- c(1, 2, 5, 10, 12, 13, 16, 19, 20)
print(binary_search(sample_vector, 13))